<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title><strong>APOSTL</strong>: Automated Processing of SAINT Templated Layouts</title>

<!-- MathJax scripts -->
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>

</head>

<body>

<h1> APOSTL Interactive Analysis </h1>

<p>Once SAINTexpress has been run, APOSTL is able to read the resulting <em>list.txt</em> file. From here APOSTL calculates normalized spectral abundance factor (NSAF) values for each prey based on the average spectra observed for each bait. Additionally, APOSTL will calculate the NSAF Score as described below. <em>Optionally</em>, APOSTL calculates the probability of a specific interaction based on prey prevalence in the <a href="http://crapome.org" target="_blank">CRAPome</a>. </p>

<hr/>
<h4> 1) List File </h4>
<p> Please specify the <em>List</em> file output from SAINTexpress. </p>

<h4> 2) Prey </h4>
<p> Please specify the <em>Prey</em> file used to generate the <em>List</em> file from SAINTexpress. </p>

<h4> 3) CRAPome </h4>
<p> <strong> Optional: </strong> Please specify the output file from <a href="http://crapome.org/?q=wk_1_1_search" target="_blank">Workflow 1</a> from the CRAPome website. This can be generated by copying and pasting the first column of your <em> Prey </em> file into the <strong> Enter Protein Identifiers </strong> field and clicking the <em> Download as tab-delimited file </em> link under your <strong> Query Results </strong>. </p>

<h4> 4) Inter File </h4>
<p> Please specify the <em>Inter</em> file used to generate the <em>List</em> file from SAINTexpress. </p>
<hr/>
<p>Once you have loaded your data into Galaxy, APOSTL will generate an HTML file with a link to your personal <strong>interactive analysis environment</strong>. To view this HTML file, click the <img src="eye_icon.png"></img> icon on the new file in your history and open the link in a new browser tab</p>

<hr/>

<p>Within an <strong>interactive analysis environment</strong>, APOSTL offers various visualization and analysis options including:</p>

<p>1) <em>Quality control</em>: Correlations between replicate runs</p>
<p>2) <em>Quality control</em>: Boxplots of selected protein across all replicates</p>
<p>3) Bubble graphs for each bait with numerous customization options:</p>

<ul>
<li>Axis and bubble scaling choices:

<ul>
<li>\(ln(NSAF) = ln(\frac{SpC/L}{\sum_{n=1}^N (SpC/L)})\)</li>
<li>\(\sum_{n=1}^N SpC\)</li>
<li>\(log_{2}(FC)\)</li>
<li>\(Saint Score\)</li>
<li>\(Log Odds Score\)</li>
<li>\(NSAF Score = ln(\frac{NSAF_T + &epsilon;}{\frac{NSAF_c} n + &epsilon;})\)</li>
</ul></li>
</ul>

<blockquote>
<p>\(SpC\) = Spectral Count (or MS1 Intensity)</p>
<p>\(L\) = Amino Acid Length</p>
<p>\(FC\) = Fold Change</p>
<p>\(&epsilon; = 1/mean(\frac{SpC}{L})_c\)</p>
<p>\(n\) = # control replicates</p>
</blockquote>

<ul>
<li><em>Optional</em>: bubble color corresponds to the CRAPome probability of a specific interaction in which an 80% cutoff is applied where prey with ≤ 80% are colored separately. Input is the respective output file from <a href="http://crapome.org/?q=wk_1_1_search" target="_blank">Workflow 1</a> from the CRAPome website.</li>
<li>Numerous cosmetic options including labels, color schemes and plotting themes</li>
</ul>

<p>4) Density plots of all baits for options specified above</p>
<p>5) Cytoscape networks of all preys passing filtering criteria displaying newly identified bait-prey interactions</p>
<p>6) Filtered data displayed as a table</p>
<p>7) KEGG pathway analysis of filtered data</p>
<p>8) Gene Ontology (GO) term analysis of filtered data</p>

<p>All graphs, images and data can be saved using the <i class="icon-download"></i> Download buttons to generate 600 dpi figures or alternatively by <kbd>RMB + Save As</kbd>.</p>

<hr/>

<h4 id="references">References</h4>

<ol>
<li><p>Teo G, Liu G, Zhang J, Nesvizhskii AI, Gingras A-C, Choi H. SAINTexpress: improvements and additional features in Significance Analysis of INTeractome software. <em>J Proteomics</em>. 2014 Apr 4;100:37–43</p></li>
<li><p>Zybailov B, Coleman MK, Florens L, Washburn MP. Correlation of relative abundance ratios derived from peptide ion chromatograms and spectrum counting for quantitative proteomic analysis using stable isotope labeling. <em>Anal Chem</em>. 2005 Oct 1;77(19):6218–24</p></li>
</ol>

<h4 id="contact-us">Contact us</h4>

<p>APOSTL support is provided by the Haura and Rix labs:</p>

<ul>
<li>Adam Borne: <a href="mailto:Adam.Borne@moffitt.org">Adam.Borne@moffitt.org</a></li>
<li>Brent Kuenzi: <a href="mailto:Brent.Kuenzi@moffitt.org">Brent.Kuenzi@moffitt.org</a></li>
<li>Paul Stewart, PhD: <a href="mailto:Paul.Stewart@moffitt.org">Paul.Stewart@moffitt.org</a></li>
</ul>

<p>Source code is available on <a href="https://github.com/bornea/APOSTL" target="_blank">Github</a></p>

</body>

</html>