<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title><strong>APOSTL</strong>: Automated Processing of SAINT Templated Layouts</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>



<!-- MathJax scripts -->
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>




</head>

<body>

<h2 id="apostl-automated-processing-of-saint-templated-layouts"><strong>APOSTL</strong>: Automated Processing of SAINT Templated Layouts</h2>

<hr/>

<p><div class="toc">
<ul>
<li><a href="#apostl-automated-processing-of-saint-templated-layouts"><strong>APOSTL: Automated Processing of SAINT Templated Layouts</strong></a><ul>
<li><a href="#overview"><strong>Overview</strong></a></li>
<li><a href="#Walkthrough"><strong>Walkthrough</strong></a></li>
<li><a href="#pre-processing"><strong>Pre-processing</strong></a></li>
<li><a href="#post-processing"><strong>Post-processing</strong></a></li>
<li><a href="#references"><strong>References</strong></a></li>
<li><a href="#contact-us"><strong>Contact us</strong></a></li>
<li><a href="#errors"><strong>Help</strong></a></li>
</ul>
</li>
</ul>
</div>
</p>

<hr/>

<h2 id="overview"> Overview</h2>

<p>APOSTL is an interactive affinity proteomics analysis software developed to reformat affinity proteomics data (both spectral counting and MS1) for input into the SAINTexpress statistical package\(^1\) and to visualize the output(s).  APOSTL was developed at H. Lee Moffitt Cancer Center &amp; Research Institute and distributed under a GNU General Public License (GPL). APOSTL is built in Python and R and integrated with SAINTexpress into a cohesive affinity proteomics data analysis package using the Galaxy framework.</p>

<h2 id="Walkthrough">Walkthrough</h2>

<p>This is a short advanced user Walkthtough. A long format Walkthrough with pictures can be found here <a href="http://apostl.moffitt.org:8080/static/APOSTL_WT.html">here</a>.</p>
<ol>
<li><p>Please create an account by clicking on the User menu button and selecting register.</p></li>
<li><p>Signed in as a user click on Shared Data menu button and select Published Histories.</p></li>
<ul><li>Here is the EGFR LFQ Scaffold and MaxQuant results files along with every step taken in prodcuing the results.</ul></li>
<ul><li>These Histories can be imported into the History panel on the Home Page by clicking on them and selecting import history in the upper right corner.</ul></li>
<ul><li>In each History there are generated APOSTL Interactive Analysis datasets which can be accessed by selecting the view data button and clicking the APOSTL Interactive Analysis link in the dataset. </ul></li>
<li><p>Click on Shared Data menu button and select Published Workflows.</p></li>
<ul><li>Here is the Workflows to be used on other datasets, simply upload a Scaffold Output/ MaxQuant Peptides file, a Bait File (This can also be created using the bait create in the preprocessing tool), and a crapome file. Then import the workflow and run it from the Workflows menu button.</ul></li>
</ol>

<h2 id="pre-processing">Pre-processing</h2>

<p>Within the <strong>Galaxy environment</strong> APOSTL is able to recognize either a Scaffold (<a href="http://www.proteomesoftware.com/products/scaffold/" target="_blank">Proteome Software</a>) <em>Samples Report</em> file or the <em>peptides.txt</em> file in the MaxQuant (<a href="http://coxdocs.org/doku.php?id=maxquant:start." target="_blank">link</a>) <em>txt</em> output folder. No modifications should be made to these files. Using the <strong>Bait Create</strong> tool, you can create your <em>bait.txt</em> file. It is important that the individual bait names match the bait names within your Scaffold or MaxQuant output. APOSTL uses the bait file to find the user&#39;s baits of interest. Additionally there is an option to make the prey file (Y/N).</p>

<p>When making a prey file, APOSTL queries a user provided FASTA database in order to extract protein amino acid lengths and gene names. This may take several minutes depending on your computer and if your Galaxy distribution is cluster enabled. Some users may want to run SAINTexpress using the same data set while changing which baits are considered test or control. It is useful to toggle <strong>Make Prey</strong> off in order to save time by circumventing this step as the same prey file can be used for both SAINTexpress runs.</p>

<p><strong>Supported Inputs</strong>:</p>

<ul>
<li>Scaffold output:

<ul>
<li><em>Samples Report</em> output</li>
</ul></li>
<li>MaxQuant output:

<ul>
<li><em>peptides.txt</em> file</li>
</ul></li>
</ul>

<blockquote>
<p><strong>Note:</strong> All files must be <em>tab delimited txt</em></p>
</blockquote>

<h2 id="post-processing">Post-processing</h2>

<p>Once SAINTexpress has been run, APOSTL is able to read the resulting <em>list.txt</em> file. From here APOSTL calculates normalized spectral abundance factor\(^2\) (NSAF) values for each prey based on the average spectra observed for each bait. Additionally, APOSTL will calculate the NSAF Score as described below. <em>Optionally</em>, APOSTL calculates the probability of a specific interaction based on prey prevalence in the <a href="http://crapome.org" target="_blank">CRAPome</a>. </p>

<hr/>

<p>Within an <strong>interactive analysis environment</strong>, APOSTL offers various visualization and analysis options including:</p>

<p>1) Correlations between replicate runs</p>
<ul><li>Axes are spectral counts or intensity values of all measured proteins for the selected replicates</ul></li>
<p>2) Boxplots of selected protein across all replicates</p>
<ul><li>Abundance refers to spectral counts or intensity values respectively</ul></li>
<p>3) Bubble graphs for each bait with numerous customization options:</p>

<ul>
<li>Axis and bubble scaling choices:

<ul>
<li>\(ln(NSAF) = ln(\frac{SpC/L}{\sum_{n=1}^N (SpC/L)})\)</li>
<li>\(\sum_{n=1}^N SpC\)</li>
<li>\(log_{2}(FC)\)</li>
<li>\(Saint Score\)</li>
<li>\(Log Odds Score\)</li>
<li>\(NSAF Score = ln(\frac{NSAF_T + &epsilon;}{\frac{NSAF_c} n + &epsilon;})\)</li>
</ul></li>
</ul>

<blockquote>
<p>\(SpC\) = Spectral Count (or MS1 Intensity)</p>
<p>\(L\) = Amino Acid Length</p>
<p>\(FC\) = Fold Change</p>
<p>\(&epsilon; = 1/mean(\frac{SpC}{L})_c\)</p>
<p>\(n\) = # control replicates</p>
</blockquote>

<ul>
<li><em>Optional</em>: bubble color corresponds to the CRAPome probability of a specific interaction in which an 80% cutoff is applied where prey with â‰¤ 80% are colored separately. Input is the respective output file from <a href="http://crapome.org/?q=wk_1_1_search" target="_blank">Workflow 1</a> from the CRAPome website.</li>
<li>Numerous cosmetic options including labels, color schemes and plotting themes</li>
</ul>

<p>4) Density plots of all baits for options specified above</p>
<ul><li>EPS file download not supported for density plots</ul></li>
<p>5) Cytoscape networks of all preys passing filtering criteria displaying newly identified bait-prey interactions</p>
<ul><li> Networks can be saved as png images or exported as JSON (for import into cytoscapejs) or Simple Interaction File (SIF) format (for direct importing into cytoscape)</li></ul>
<p>6) Filtered data displayed as a table</p>
<p>7) KEGG pathway analysis of filtered data</p>
<p>8) Gene Ontology (GO) term analysis of filtered data</p>

<p>All graphs, images and data can be saved using the <i class="icon-download"></i> Download buttons or alternatively by <kbd>RMB + Save As</kbd> to generate 600 dpi figures.</p>

<hr/>

<p>Within the <strong>Galaxy environment</strong>, APOSTL offers several other analysis options including:</p>

<p>1) Protein-protein interaction networks querying known PPI as annotated in <a href="http://consensuspathdb.org" target="_blank">ConsensusPathDB</a> following filtering for simple cytoscape import using the <em>import network from file</em> option.</p>

<p>2) Dot plot and clustering analysis from the the <a href="http://prohitstools.mshri.on.ca" target="_blank">Prohits Visualization Tool</a>\(^3\)</p>

<h2 id="references">References</h2>

<ol>
<li><p>Teo G, Liu G, Zhang J, Nesvizhskii AI, Gingras A-C, Choi H. SAINTexpress: improvements and additional features in Significance Analysis of INTeractome software. <em>J Proteomics</em>. 2014 Apr 4</p></li>
<li><p>Zybailov B, Mosley AL, Sardiu ME, Coleman MK, Florens L, Washburn MP. Statistical Analysis of Membrane Proteome Expression Changes in <em> Saccharomyces cerevisiae. J Proteome Res</em>. 2006 Apr 10</p></li>
<li><p>Knight JD, Liu G, Zhang JP, Pasculescu A, Choi H, Gingras AC. A web-tool for visualizing quantitative protein-protein interaction data. <em> Proteomics</em>. 2015 Jan 19</p></li>
</ol>

<h2 id="contact-us">Contact us</h2>

<p>APOSTL support is provided by the Haura and Rix labs:</p>

<ul>
<li>Adam Borne: <a href="mailto:Adam.Borne@moffitt.org">Adam.Borne@moffitt.org</a></li>
<li>Brent Kuenzi: <a href="mailto:Brent.Kuenzi@moffitt.org">Brent.Kuenzi@moffitt.org</a></li>
<li>Paul Stewart, PhD: <a href="mailto:Paul.Stewart@moffitt.org">Paul.Stewart@moffitt.org</a></li>
</ul>

<p>Source code is available on <a href="https://github.com/bornea/APOSTL" target="_blank">Github</a></p>

<h2 id="errors">Help </h2>
<p>Below are some occasional errors that we have encountered while testing APOSTL with solutions:</p>

<h5>General Error</h5>
<p><code style="color:red"><strong>Error:</strong> length of 'dimnames' [2] not equal to array extent</code> This error is usually due to the user filtering data such that it has removed all the points in your data. Adjust your filtering to resolve.</p>

<h5>Cytoscape Network</h5>
<p>Occasionally when rendering the Cytoscape network in the browser, node shapes can become white when changing them. This issue has not been fully resolved in the Beta Release of this tool. To fix this issue currently, toggle different shapes in your network until the shape you want remains.</p>

<h5>KEGG Pathway Analysis</h5>
<p><code style="color:red"><strong>Error:</strong> invalid request, server returned Success: (200) OK (http://rest.kegg.jp/link/hsa/pathway)</code> This is usually due to your R Bioconductor installation being out of date. It can usually be resolved by running the
<code> BiocInstaller::biocLite()</code> command in R on the machine hosting the Shiny server and Galaxy.</p>

</body>

</html>
